test/e2e/runner/cli_provisioner.go

```diff
@@ -113,13 +113,15 @@ func (cli *CLIProvisioner) provision() error {
 	}
 	os.Setenv("NAME", cli.Config.Name)
 
-	outputPath := filepath.Join(cli.Config.CurrentWorkingDir, "_output")
-	if !cli.Config.UseDeployCommand {
-		publicSSHKey, err := createSaveSSH(outputPath, cli.Config.Name+"-ssh")
-		if err != nil {
-			return errors.Wrap(err, "Error while generating ssh keys")
+	if cli.Config.PublicSSHKey == "" || cli.Config.PrivateSSHKeyPath == "" {
```

Comment:
Users may see unexpected hard-to-troubleshoot behavior if they set one out of the two variables.

I would validate that either none or both are set.

Also, this code should also set `PUBLIC_SSH_KEY`, probably not good to assume that this is always going to be called from `cluster.sh`